src/intents.js

```diff
@@ -0,0 +1,61 @@
+const intentClass = 'coz-intent'
+
+function getUrl (intent, temporaryMockedIntentUrl) {
+  return Promise.resolve(temporaryMockedIntentUrl)
+}
+
+function injectService (url, element, intent) {
+  const document = element.ownerDocument
+  if (!document) throw new Error('Cannot retrieve document object from given element')
+
+  const window = document.defaultView
+  if (!window) throw new Error('Cannot retrieve window object from document')
+
+  const iframe = document.createElement('iframe')
+  iframe.setAttribute('src', url)
+  iframe.classList.add(intentClass)
+  element.appendChild(iframe)
+
+  // Keeps only http://domain:port/
+  const serviceOrigin = url.split('/', 3).join('/')
+
+  return new Promise((resolve, reject) => {
+    let handshaken = false
+    const messageHandler = (event) => {
+      if (event.origin !== serviceOrigin) return
+
+      if (event.data === 'intent:ready') {
+        handshaken = true
+        return event.source.postMessage(intent.data, event.origin)
+      }
+
+      window.removeEventListener('message', messageHandler)
+
+      if (event.data === 'intent:error') {
+        return reject(new Error('Intent error'))
+      }
+
+      return handshaken
+        ? iframe.parentNode.removeChild(iframe) && resolve(event.data)
+        : reject(new Error('Unexpected handshake message from intent service'))
+    }
+
+    window.addEventListener('message', messageHandler)
+  })
+}
+
+export function start (client, intent, element, temporaryMockedIntentUrl) {
+  ['action', 'type'].forEach((property) => {
```

Comment:
@y-lohse I thank about it. Actually what bothers me is to have `element` as first parameter, I think `intent`, or intent's properties have to be in first position. But then, I think that we got to much params, with four related to the intent, and one last (`element`), absolutely not. So maybe we could make something like:
```javascript
cozy.client.intents.create('PICK', 'io.cozy.files).start(element).then(document => {})
//or
cozy.client.intents.create('EDIT', 'io.cozy.contact).then(intent => cozy.client.intents.start(intent, element).then(document => {})
```
By the way, a time will come when we will need to create an intent in DB. Maybe this syntax will be the best one.